### PR TITLE
plugin: add ceilToDecimals() method to round floats to specific places

### PR DESCRIPTION
**Pull request by schinkowski**

### CODE REVIEW

File: Plugin~/Src/MeshSyncTest/TestMath.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	auto difference = val - expected;
+
+	return fabs(difference) < muEpsilon;
+}
+
+TestCase(TestRoundToDecimal)
+{
+	// 2 places:
+	// Round down
+	assert(compare(roundToDecimals(mu::tvec3<float>{ 1.121f, 2.342f, 3.453 }), { 1.13f, 2.35f, 3.46f }) && "Error rounding tvec3<float> to 2 decimal places");
```
Comment (SonnyCampbellUnity) [inline Plugin~/Src/MeshSyncTest/TestMath.cpp]: This doesn't look like it's actually rounding down? Is the comment incorrect or am I misunderstanding something?

1.121 -> 1.13 ?

File: Plugin~/Src/MeshUtils/Include/MeshUtils/muMath.h [hunk 2]
```diff
@@ -1651,4 +1651,12 @@ static inline void enumerate(A1& a1, A2& a2, A3& a3, const Body& body)
         body(a1[i], a2[i], a3[i]);
 }
 
+template
+<typename T>
+T roundToDecimals(T val, float decimalPlaces = 2) {
```
Comment (sindharta) [inline Plugin~/Src/MeshUtils/Include/MeshUtils/muMath.h]: `decimalPlaces` should be an int

File: Plugin~/Src/MeshUtils/Include/MeshUtils/muMath.h [hunk 3]
```diff
@@ -1651,4 +1651,12 @@ static inline void enumerate(A1& a1, A2& a2, A3& a3, const Body& body)
...
+    assert(decimalPlaces >= 1);
+    float roundingFactor = std::pow(10, decimalPlaces);
```
Comment (sindharta) [inline Plugin~/Src/MeshUtils/Include/MeshUtils/muMath.h:1658]: using `double` and then returning as `T` may be more accurate ?

File: Plugin~/Src/MeshSyncTest/TestMath.cpp (hunk 1 above)
Comment (schinkowski) [reply Plugin~/Src/MeshSyncTest/TestMath.cpp]: It always rounds up, I tried std::round but it also has floating point inaccuracies and doesn't create floats that are suitable for hashing. ceil works better and is good enough for this use case. I've removed those comments.

File: Plugin~/Src/MeshUtils/Include/MeshUtils/muMath.h (hunk 2 above)
Comment (schinkowski) [reply Plugin~/Src/MeshUtils/Include/MeshUtils/muMath.h]: Fixed

File: Plugin~/Src/MeshUtils/Include/MeshUtils/muMath.h (hunk 3 above)
Comment (schinkowski) [reply Plugin~/Src/MeshUtils/Include/MeshUtils/muMath.h:1658]: This needs to work with types that cannot be cast to double like float3, etc. Also, this rounding function is only to be used for hashing, accuracy isn't really important. We just need to clamp floats that are very similar to the same number so their hashes are the same.

**Review by SonnyCampbellUnity — APPROVED** [review]

File: Plugin~/Src/MeshSyncTest/TestMath.cpp (hunk 1 above)
Comment (sindharta) [reply Plugin~/Src/MeshSyncTest/TestMath.cpp]: If we are using ceil, can we please rename the function to indicate that it is using `ceil()` ? `ceilToDecimal` perhaps ?

Another alternative would be implement the correct rounding, meaning that 
1.121  -> 1.12
1.127 -> 1.13

by using typecasting. See [this page](https://java2blog.com/round-to-2-decimal-places-cpp/#Using_the_ceil_function_to_round_to_2_decimal_places_in_C) for example.

I know that in our current use case, we don't necessarily need this correctness, but someone may use it in the future for other use cases, so making it clear if it does `ceil()` or not will make it less misleading.

Comment (schinkowski) [reply Plugin~/Src/MeshSyncTest/TestMath.cpp]: I've renamed it.

**Pull request merged**